stdlib/public/core/MemoryLayout.swift

```diff
@@ -44,9 +41,29 @@ public enum MemoryLayout<T> {
 }
 
 extension MemoryLayout {
+  /// Returns the contiguous memory footprint of `T`.
+  ///
+  /// Does not include any dynamically-allocated or "remote" storage. In
+  /// particular, `MemoryLayout(ofValue: x).size`, when `x` is a class instance,
```

Comment:
Thanks! Nate Cook has just cleaned up the documentation for the existing bits of MemoryLayout quite a bit, so I'll either rebase or submit a follow-up patch and incorporate a correction.